pkg/3scale/amp/component/system_options.go

```diff
@@ -92,6 +92,8 @@ type SystemOptions struct {
 	IncludeOracleOptionalSettings bool
 
 	BackendServiceEndpoint string `validate:"required"`
+
+	Namespace string `validate:"required"`
```

Comment:
A thought about this, I wonder whether in the future seeing this `Namespace` generic option (in all options components) might be confusing to people looking at the components code (like, "why is the Namespace an option and only being used in monitoring objects?"). Right now it is only used for monitoring-related objects. All the other objects of the components do not make use of this namespace because the Namespace is set outside.

Reply:
Maybe it deserves a comment. Objects created are namespaced. However, objects includes labels, rules and expressions that need namespace filtering because they are "global" once imported to the prometheus or grafana services.